### ISSUE TITLE
[2019.09] PHP Warnings

### ISSUE THREAD

**Issue by MrPetovan** [friendica/friendica#7298]
Please report here any PHP Warning message obtained running the `2019.06` stable or the `2016.09-dev` version.

**Comment by AlfredSK** [comment]
Hello!

`PHP Warning:  array_merge(): Argument #2 is not an array in .../src/Network/Probe.php on line 354`

2016.09-dev

**Comment by AlfredSK** [comment]
^^^^^
This is currently flooding my PHP log with 2-3 new entries every second.

**Comment by copiis** [comment]
Develop

`PHP Warning:  unlink(/home/users/meld/tmp/MELD.de/worker_process.sem): No such file or directory in /src/Core/Lock/SemaphoreLock.php on line 72`

**Comment by nupplaphil** [comment]
> Develop
> 
> `PHP Warning: unlink(/home/users/meld/tmp/MELD.de/worker_process.sem): No such file or directory in /src/Core/Lock/SemaphoreLock.php on line 72`

Caused by #7515 , strange thing that this can happen, I'll add a check :-)

**Comment by copiis** [comment]
develop

`PHP Warning:  ftok(): ftok() failed - No such file or directory in /src/Core/Lock/SemaphoreLock.php on line 29`

**Comment by nupplaphil** [comment]
> 
> 
> develop
> 
> `PHP Warning: ftok(): ftok() failed - No such file or directory in /src/Core/Lock/SemaphoreLock.php on line 29`

@copiis Did you update the latest git version? Just to be sure, because I don't find any possibility to call this method without a file.. (

**Comment by copiis** [comment]
No i did a git pull now and afther that this
`
PHP Warning:  unlink(/home/users/meld/tmp/MELD.de/worker_process.sem): No such file or directory in /src/Core/Lock/SemaphoreLock.php on line 73`

**Comment by copiis** [comment]
And this still keeps coming`
PHP Warning:  ftok(): ftok() failed - No such file or directory in /src/Core/Lock/SemaphoreLock.php on line 29`

**Comment by nupplaphil** [comment]
Thanks.. I'll check it again..

**Comment by copiis** [comment]
`PHP Warning:  Invalid argument supplied for foreach() in /src/Model/GContact.php on line 758`

**Comment by noplanman** [comment]
When selecting `Reload active themes` (`/admin/themes?action=reload`)
```
Warning: Illegal string offset 'name' in /var/www/html/src/Module/Admin/Themes/Index.php on line 29
Warning: Illegal string offset 'name' in /var/www/html/src/Module/Admin/Themes/Index.php on line 30
Warning: Illegal string offset 'name' in /var/www/html/src/Module/Admin/Themes/Index.php on line 29
Warning: Illegal string offset 'name' in /var/www/html/src/Module/Admin/Themes/Index.php on line 30
Warning: Illegal string offset 'name' in /var/www/html/src/Module/Admin/Themes/Index.php on line 29
Warning: Illegal string offset 'name' in /var/www/html/src/Module/Admin/Themes/Index.php on line 30
Warning: Illegal string offset 'name' in /var/www/html/src/Module/Admin/Themes/Index.php on line 29
Warning: Illegal string offset 'name' in /var/www/html/src/Module/Admin/Themes/Index.php on line 30
Warning: Cannot modify header information - headers already sent by (output started at /var/www/html/src/Module/Admin/Themes/Index.php:29) in /var/www/html/src/Core/System.php on line 250
```

**Comment by copiis** [comment]
`PHP Warning:  sem_acquire(): failed to acquire key 0x66010697: Identifier removed in a/src/Core/Lock/SemaphoreLock.php on line 43`

**Comment by copiis** [comment]
PHP Notice:  Undefined index: url in addon/twitter/twitter.php on line 624

**Comment by copiis** [comment]
`PHP Warning:  file_put_contents(/home/users/meld/www/friendica/proxy/https://meld.de/photo/12364994125ac9dd5d3fe12259626166-6.png?ts=1523178845): failed to open stream: No such file or directory in /src/Module/Proxy.php on line 116\nPHP message: PHP Warning:  file_put_contents(/home/users/meld/www/friendica/proxy/https://meld.de/photo/12364994125ac9dd5d3fe12259626166-6.png?ts=1523178845:thumb): failed to open stream: No such file or directory in /src/Module/Proxy.php on line 134\n', referer: https://meld.de/search?tag=before`

**Comment by AlfredSK** [comment]
The file path looks wrong.

**Comment by copiis** [comment]
PHP Warning:  file_put_contents(/home/users/meld/www/friendica/proxy/https://meld.de/photo/37030339515aca20968ed9c495401044-6.png?ts=1523196054): failed to open stream: No such file or directory in /home/users/meld/www/friendica/src/Module/Proxy.php on line 116\nPHP message: PHP Warning:  file_put_contents(/home/users/meld/www/friendica/proxy/https://meld.de/photo/37030339515aca20968ed9c495401044-6.png?ts=1523196054:thumb): failed to open stream: No such file or directory in /home/users/meld/www/friendica/src/Module/Proxy.php on line 134\n'

> The file path looks wrong.

Witch one do you mean?

**Comment by adrianheine** [comment]
`PHP Warning:  copy(/home/agora/friendica/.htaccess): failed to open stream: Permission denied in /home/agora/friendica/src/Worker/Cron.php on line 114` (mostly readonly git checkout served by nginx)

**Comment by tobiasd** [comment]
There was a change in the repository. We now have a `.htaccess-dist` (or something) in the repository that you have to copy over to `.htaccess` so you can add local changes to it without colliding revisions in the repository. Maybe that is the issue in your case?

**Comment by adrianheine** [comment]
Yes, but since I use nginx I didn't copy the file and don't need it.

**Comment by tobiasd** [comment]
On Sat, 14 Sep 2019 00:59:02 -0700
Adrian Heine né Lang <notifications@github.com> wrote:

> Yes, but since I use nginx I didn't copy the file and don't need it.

I see :-) nice catch then!

**Comment by MrPetovan** [comment]
We have no way of knowing which web server Friendica is served with, but we can check if we have the required permission to perform this file copy. Thanks for the report.

**Comment by AlfredSK** [comment]
Current RC:

```
count(): Parameter must be an array or an
object that implements Countable in .../src/Module/Profile/Contacts.php on line 49
```

**Comment by AlfredSK** [comment]
```PHP Warning:  Use of undefined constant visitor - assumed 'visitor' (this will throw an Error in a future version of PHP) in .../mod/redir.php on line 159```

PHP 7.2

I'm not sure if that one popped up before or after the latest change to 'redir'.

**Comment by annando** [comment]
@AlfredSK found and is fixed in my open PR

**Comment by nupplaphil** [comment]
Closes in favor of #7676
